syntax/printer.go

```diff
@@ -269,16 +269,17 @@ func (p *Printer) space() {
 }
 
 func (p *Printer) spacePad(pos Pos) {
-	if p.cols.lineStart {
-		// Never add padding at the start of a line, since this may
+	if p.cols.lineStart && (!p.keepPadding || p.indentSpaces == 0) {
```

Comment:
I *think* you can avoid the keepPadding check here, because we didn't have it before. I think that, when keepPadding is false, the column counter isn't running, so p.cols.column==0 and the rest of the logic doesn't trigger.